consumer/consumer.go

```diff
@@ -94,9 +95,15 @@ func NewWithSaramaConfig(
 		return nil, err
 	}
 
-	payloadTrackerProducer, err := producer.New(brokerCfg)
+	payloadTrackerProducer, err := producer.CreatePayloadTrackerProducer(brokerCfg)
```

Comment:
Maybe you can call it `NewPayloadTrackerProducer` instead of `CreatePayloadTrackerProducer`?